src/auth/authclient.ts

```diff
@@ -14,12 +14,18 @@
  * limitations under the License.
  */
 
-import * as request from 'request';
+import axios, {AxiosPromise, AxiosRequestConfig, AxiosResponse} from 'axios';
 
 import {DefaultTransporter} from '../transporters';
 
 import {Credentials} from './credentials';
 
+export interface BodyResponse {
+  // tslint:disable-next-line no-any
+  body: any;
```

Comment:
Is this the same value as `res.data`? In that case let the users dig it out. You're making a semver major change regardless, and abbreviating `res.data` to `body` isn't that big a win. That gets rid of this whole interface.

Secondly, instead of using `any`, you can use generics similarly to how `axios` itself is [doing](https://github.com/axios/axios/blob/d7f021b8d4cc50bfa0653011bc02452d234d1255/index.d.ts#L51).

Reply:
Okay, so I think I understand your motivation. This way, the new API continues working without requiring changes from users, *in most cases*. The tests themselves are easy to migrate.

This is very dangerous. Unless you are 100% compatible you should not pretend to be compatible. The users need to migrate; by making their code work in most cases will set them up for major sadness in the obscure cases where it happens to not work.

If this is not a duck, it should not walk like a duck, nor talk like a duck.

Reply:
I don't think this particular API is in heavy use by users in any case, so we should not try to optimize for 'ease for migration'.

Reply:
If we can switch out request for axios, AND add support to async APIs, all while maintaining 90% compatibility - I'd really like to do it.  For most users, they're not going to need to make any changes.  For the users that are really deeply using the API, the only real differences will be:
- Inputs to the various request methods (we talked about throwing when they look like non-axios supported objects)
- The response objects in the callback.  In most cases from what I can tell, people just grab the body and work with that.  In the event that they actually need the response object, they're going to figure out pretty quickly that it's different - all the property names are different, even statusCode.  

I know we're in the uncanny valley of compatibility, but I think we can *really* reduce the pain of this upgrade for > 90% of dependents.

Reply:
> I know we're in the uncanny valley of compatibility, but I think we can really reduce the pain of this upgrade for > 90% of dependents.

You're changing behavior in a very subtle way. 90% of users will probably not see any pain, but the pain to the remaining 10% could be substantial. For example, `request` provides only network errors via `err`, whereas axios considers 5xx responses to be errors too.

The evidence that would be required to convince me is a sampling of dependent modules that demonstrates that a "no-thinking-required migration" in fact produces the behavior those modules expected.

Reply:
Nah, you convinced me :) I'll try it out.